### ISSUE TITLE
Cannot give pattern-matching lambda in abstract setting

### ISSUE THREAD

**Issue by nad** [agda/agda#4007]
```agda
abstract

  data D : Set where
    c : D

  works : D → D
  works = λ { c → c }

  fails : D → D
  fails = {!λ { c → c }!}
```
Agda complains when I try to "give" the given expression in the hole above:
```
Cannot split on abstract data type D
when checking that the pattern c has type D
```

**Comment by nad** [comment]
Here is a related test case:
```agda
abstract

  data D : Set where
    c : D

  works : D
  works = λ { → c }

  fails : D
  fails = {!λ { → c }!}
```
```
Constructor c is abstract, thus, not in scope here
when checking that the expression c has type D
```

**Comment by andreasabel** [comment]
Weird, since `giveExpr` does enter the context of the metavariable using `withMetaInfo`, which sets the `TCEnv` containing the `envAbstractMode` flag.

**Comment by andreasabel** [comment]
Testing giving
(a) `\ { x -> c }` 
(b) `\ { c -> c }`

on  previous Agda versions:
- 2.4.2.4 does not even accept `works`.
- 2.4.2.5 - 2.5.1.1 accept (a), complain for (b) about `type mismatch`.
- 2.5.2 (a) complains `Not in scope: Issue4007.D.c (did you mean 'c'?)` :D, (b) type mismatch.
- 2.5.3 has same behavior as current Agda.

So, 2.5.2 introduced a regression for (a) while (b) may have never worked.

**Comment by andreasabel** [comment]
Puh, a >3 hour bug hunt.  Everything was set up already for this to work correctly.
- `checkExtendedLambda` respects the `defAbstract` information coming with the `A.DefInfo` of the `A.ExtendedLam`. 
- `scopeCheckExtendedLam` puts the information about abstract in from the `TCEnv` when creating the extended lambda.
- `giveExpr` runs in the context of the interaction meta; thus, in `AbstractMode`.

However, `BasicOps.parseExprIn` did not invoke `withMetaInfo` when calling the scope checker, thus, the information about `AbstractMode` was lost here in an innocuous place.  (I was looking in all the wrong places.)

Thus, the final fix is half a line.

Moral of the story: `abstract` is underrepresented in the test suite, otherwise this error wouldn't have persisted when the underlying problem #1677 was (at least partially) fixed in 2.4.2.5, see d208ebfb7ad5c37b65b4fa22310c6162d01650fa.
